egui_glium/src/painter.rs

```diff
@@ -319,3 +319,22 @@ impl Painter {
         }
     }
 }
+
+impl epi::NativeTexture for Painter {
+    type Texture = glium::texture::srgb_texture2d::SrgbTexture2d;
+
+    fn bind_native_texture(&mut self, native: Self::Texture) -> egui::TextureId {
+        self.register_glium_texture(native)
+    }
+
+    fn replace_texture_ref(&mut self, id: egui::TextureId, replacing: Self::Texture) {
+        if let egui::TextureId::User(id) = id {
+            if let Some(Some(user_texture)) = self.user_textures.get_mut(id as usize) {
+                *user_texture = UserTexture {
+                    pixels: vec![],
+                    gl_texture: Some(replacing),
+                };
+            }
+        }
+    }
+}
```

Comment:
I don't see the point of the `epi::NativeTexture` trait - the point of `epi` is to be able to write code that is the same on native and web, but this trait has a different `Texture` on different platforms, and assumes the user will be using native WebGL/glium functions to create it.

I guess maybe it can be used to write some sort of generic texture manager?

Reply:
yes i think we can write backend independent texture manager.